src/item-navigator.ts

```diff
@@ -154,13 +168,27 @@ export class ItemNavigator
     `;
   }
 
+  slotChange(e: Event, type: 'header' | 'main'): void {
+    const slottedContent = (e.target as HTMLSlotElement).assignedNodes()?.[0] as HTMLElement;
+
+    this.dispatchEvent(
+      new CustomEvent('slotChange', {
+        detail: { slot: slottedContent, type },
```

Comment:
Q: How is the `slottedContent` being consumed? This feels a bit weird to be sending the content as an event detail property.

Reply:
thanks for asking. this is for emitting the content found is just for reference for the `slotChange` event being dispatched.